package-res/resources/web/pentaho/type/Context.js

```diff
@@ -155,15 +240,63 @@ define([
     },
 
     /**
-     * Gets the instance constructor of a type.
+     * Gets, asynchronously, the **configured instance constructor** instance constructor of a type.
+     *
+     * For more information on the `typeRef` argument,
+     * please see [UTypeReference]{@link pentaho.type.spec.UTypeReference}.
+     *
+     * This method can be used even if a generic type metadata specification references
+     * non-standard types whose modules have not yet been loaded by the AMD module system.
+     *
+     * An error is thrown when:
```

Comment:
The analysis above seems reasonable, mostly because it adheres to the principle of a single-source-of-truth. The basic difference is that sometimes the truth is in the code, sometimes it is in the documentation.

The hybrid cases `argInvalid` and `operInvalid` are circunstancial and probably too complex  to be avoided by resorting to documentation only.

As for `@throws`, we should probably include them all because in a way it's one kind of output.
Devs invoking the code in a try-catch block would probably need to know which exceptions are there.
A block of `@throws` has the potential of becoming easily unreadable.